clu/src/lib.rs

```diff
@@ -130,6 +133,32 @@ fn linux_init(config: Arc<RwLock<settings::RitaSettingsStruct>>) -> Result<(), E
         },
     }
 
+    match device_option {
+        Some(existing_device) => {
+           info!("Device name is {}", existing_device); 
+        } 
+        None => {
+            info!("No device name was found, reading from /etc/althea-firmware-release");
+            let mut f = File::open("/etc/althea-firmware-release")?;
+            let mut contents = String::new();
+            f.read_to_string(&mut contents)?;
+
+            for line in contents.lines() {
+                if line.starts_with("device:") {
+                    let mut array = line.split(" ");
+                    array.next();
+                    match array.next() {
+                        Some(device) => {
+                            info!("Setting device name to {}", device);
+                            network_settings.device = Some(device.to_string());
+                        },
+                        None => warn!("Unable to get device from /etc/althea-firmware-release"),
```

Comment:
Wouldn't it make sense to break from this loop in this case? Is this file guaranteed to be present, and when we can't parse it it is a serious error?

Reply:
this file is pretty much garunteed to be present, but not finding it should not be a serious error. Also yes we should break the for loop once we find it.